debezium-core/src/test/java/io/debezium/transforms/ByLogicalTableRouterTest.java

```diff
@@ -196,6 +196,30 @@ public void testBrokenTopicReplacementConfigurationEmptyValue() {
         subject.configure(props);
     }
 
+    @Test
+    public void testKeyNullValue() {
+        final ByLogicalTableRouter<SourceRecord> router = new ByLogicalTableRouter<>();
+        final Map<String, String> props = new HashMap<>();
+
+        props.put("topic.regex", "(.*)customers_shard(.*)");
+        props.put("topic.replacement", "$1customers_all_shards");
+        props.put("key.field.name", "shard_id");
+        props.put("key.field.regex", "(.*)customers_shard_(.*)");
+        props.put("key.field.replacement", "$2");
+        router.configure(props);
+
+        SourceRecord record1 = new SourceRecord(
+                new HashMap<>(), new HashMap<>(), "mysql-server-1.inventory.customers_shard_1", null, null, null, null
+        );
+
+        SourceRecord transformed1 = router.apply(record1);
+        assertThat(transformed1).isNotNull();
+        assertThat(transformed1.topic()).isEqualTo("mysql-server-1.inventory.customers_all_shards");
+
+        assertThat(transformed1.keySchema()).isNull();
```

Comment:
So after thinking some more, I've come to realize that the table name should be added to the key, in order to be consistent with [what we do](https://debezium.io/docs/configuration/topic-routing/#key_fields) for table that have a PK:

> To address the concern of uniqueness across all the original tables discussed above, one more field for identifying the original (physical) table is inserted into the key structure of the change events. By default, this field is named dbzphysicalTableIdentifier and has the original topic name as its value.

Only that in the case of a table without PK, that table field would be the _only_ one in the new key structure. So naturally there wouldn't be uniqueness guaranteed, but that's just the consequence of not having a PK, and the added table identifier at least allows to keep records from different tables apart. I think that'd be the most logical thing to do for DBZ-1086; and if you also need the database name as hinted in the JIRA issue, you'd get that from the source info block (see DBZ-1087).

Does that make sense?

Reply:
It makes sense, but there is one thing which we should take into consideration - if Kafka message key is null then Kafka pick random broker partition for it. In case when all messages from one table will have the same key they will be put into the same partition and it will affect performance.

So, it looks like, null is the better option for us.

Reply:
Hum, hum, good point. So in that light the PR seems good as is then. And you'd obtain the DB name from the source info (if you're on MySQL or it would have to be implemented in the connector you're using (DBZ-1087).

Reply:
Yeah, I’m using MySQL so this PR solves all my problems)

Reply:
Ok, cool.